replica/table.cc

```diff
@@ -2039,14 +2063,15 @@ future<> compaction_group::stop() noexcept {
     if (_async_gate.is_closed()) {
         co_return;
     }
-    co_await _async_gate.close();
+    auto closed_gate_fut = _async_gate.close();
```

Comment:
better use `coroutine::as_future(_async_gate.close());`, similar to `flush_future` just below.

That said, why closing the gate in the background is safe?